src/index.ts

```diff
@@ -89,21 +93,43 @@ export default class ReactTSTarget extends Target {
             return $fieldType(value);
         });
 
-        engine.registerHelper('ifValueType', (type, options) => {
-            if ((type?.type || type?.ref) && type?.type?.toLowerCase() !== 'void' && type?.ref?.toLowerCase() !== 'void') {
-                return Template.SafeString(options.fn());
+        engine.registerHelper('ifValueType', (type: TypeLike, options: HelperOptions) => {
+            if (
+                (typeof type === 'string' && type !== 'void') ||
+                (typeof type !== 'string' && typeName(type) !== 'void')
+            ) {
+                return Template.SafeString(options.fn(this));
             }
             return Template.SafeString('');
         });
 
+        engine.registerHelper('ifVoidType', (type: TypeLike, options: HelperOptions) => {
+            // Check if type has a property 'string' (which it would have if it is a SafeString)
+            if (type && typeof type === 'object' && 'string' in type) {
+                type = type.string as string;
+            }
+
+            if (
+                (typeof type === 'string' && type === 'void') ||
+                (typeof type !== 'string' && typeName(type) === 'void')
```

Comment:
typeName will actually accept a "TypeLike" argument - you shouldn't need the typeof checks

Reply:
The `TypeLike` in codegen-target is not the same type as `TypeLike` in schemas. The `TypeLike` type the typeName function expects is not the same as is used in the codegen-target. 

```
export type TypeLike = string | {
    type?: string;
    ref?: string;
};
```
vs
```
export interface TypeLike {
    type?: string;
    ref?: string;
}
```